app/models/manageiq/providers/vmware/infra_manager/refresh_parser.rb

```diff
@@ -1075,7 +1075,7 @@ def self.vm_inv_to_disk_hashes(inv, storage_uids, storage_profile_by_disk_mor =
           new_result = {
             :device_name     => device.fetch_path('deviceInfo', 'label'),
             :device_type     => device_type,
-            :controller_type => controller_type,
+            :controller_type => %w(floppy cdrom-raw cdrom-image).include?(device_type) ? vm_mor : controller_type,
```

Comment:
I don't think we can override `controller_type` to be the vm ems_ref, if you need a vm parent ID then lets add a column for that.